src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.HelperFunctions.cs

```diff
@@ -110,21 +112,31 @@ private static T CreateBlockRequestInternal<T>(ExecutionPayload parent, Address
             {
                 blockRequestV3.DataGasUsed = dataGasUsed;
                 blockRequestV3.ExcessDataGas = excessDataGas;
+                blockRequestV3.ParentBeaconBlockRoot = beaconParentBlockRoot;
+
+                BeaconBlockRootPrecompile.Instance.SetupBeaconBlockRootPrecompileState(spec.IsEip4788Enabled, state, blockRequestV3.ParentBeaconBlockRoot, blockRequestV3.Timestamp);
+                state.Commit(spec);
+                state.CommitTree(blockRequestV3.BlockNumber);
+
+                state.RecalculateStateRoot();
+                blockRequestV3.StateRoot = state.StateRoot;
```

Comment:
This should be done only in CreateBlockRequestV3 not in generic one.

Reply:
Also potentially you don't need to pass spec here. But that depends on the test.